pkg/cli/cmd/radinit/recipe.go

```diff
@@ -52,112 +70,88 @@ func NewDevRecipeClient() DevRecipeClient {
 // GetDevRecipes is a function that queries a registry for recipes with a specific tag and returns a map of recipes.
 // If an error occurs, an error is returned.
 func (drc *devRecipeClient) GetDevRecipes(ctx context.Context) (map[string]map[string]corerp.RecipePropertiesClassification, error) {
-	reg, err := remote.NewRegistry(DevRecipesRegistry)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create client to registry %s -  %s", DevRecipesRegistry, err.Error())
-	}
+	logger := ucplog.FromContextOrDiscard(ctx)
 
 	// The tag will be the major.minor version of the release.
 	tag := version.Channel()
 	if version.IsEdgeChannel() {
 		tag = "latest"
 	}
 
-	// Temporary solution to get all repositories.
-	// The issue is that if RepositoryListPageSize is not specified the default is 100.
-	// We have 104 repositories in the registry as of 12 Oct 2023. That is why processRepositories
-	// function was being called twice and the second call was overwriting all the recipes.
-	// TODO: Remove this once we have a better solution.
-	reg.RepositoryListPageSize = 1000
-
-	recipes := map[string]map[string]corerp.RecipePropertiesClassification{}
-
-	// if repository has the correct path it should look like: <registryPath>/recipes/<category>/<type>:<tag>
-	// Ex: ghcr.io/radius-project/recipes/local-dev/rediscaches:0.20
-	// The start parameter is set to "radius-rp" because our recipes are after that repository.
-	err = reg.Repositories(ctx, "", func(repos []string) error {
-		// validRepos will contain the repositories that have the requested tag.
-		validRepos := []string{}
-		for _, repo := range repos {
-			r, err := reg.Repository(ctx, repo)
-			if err != nil {
-				continue
-			}
+	validRepos := []string{}
+	for _, recipe := range availableDevRecipes() {
+		repoPath := fmt.Sprintf("%s%s", RecipeRepositoryPrefix, recipe)
+		repo, err := remote.NewRepository(repoPath)
+		if err != nil {
+			// This shouldn't cancel the `rad init` flow.
+			logger.Error(err, fmt.Sprintf("failed to create client to repository %s", repoPath))
```

Comment:
If the recipes don't get downloaded correctly, the first time and `rad init` still completes, is the idea that the users will re-run the command to get the dev recipes? Just wondering how that case would work since it seems like they'd just keep running into issues downloading the recipes if this still fails

Reply:
Nvm, I see what this is try to do by only adding the valid repos